packages/studio/src/components/Viewport/ViewportMenu.tsx

```diff
@@ -0,0 +1,67 @@
+import { useCallback } from "react";
+import { EXPANDED_VIEWPORTS, Viewport } from "./defaults";
+import useStudioStore from "../../store/useStudioStore";
+
+interface optionProps extends Viewport {
+  afterSelect: () => void;
+}
+
+export default function ViewportMenu(props: {
+  closeMenu: () => void;
+}): JSX.Element {
+  const { closeMenu } = props;
+
+  return (
+    <div className="absolute z-20 rounded bg-white text-sm text-gray-700 shadow-lg">
+      <Selector afterSelect={closeMenu} />
+    </div>
+  );
+}
+
+function Selector(props: { afterSelect: () => void }) {
+  const { afterSelect } = props;
+
+  return (
+    <div className="flex flex-col items-start py-1">
+      {Object.values(EXPANDED_VIEWPORTS).map((val) => {
+        return <Option afterSelect={afterSelect} {...val} />;
+      })}
+    </div>
+  );
+}
+
+function Option(props: optionProps) {
+  const [setViewportDimensions] = useStudioStore((store) => [
+    store.pagePreview.setViewportDimensions,
+  ]);
+  const { afterSelect } = props;
+  const viewportOption = props.name;
+
+  const handleSelect = useCallback(() => {
+    afterSelect?.();
```

Comment:
afterSelect is required in `optionProps` so you can get rid of the optional chaining